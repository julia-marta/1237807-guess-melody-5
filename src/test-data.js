import React from 'react';
import PropTypes from "prop-types";

export const noop = () => {};

export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [{
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `blues`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `jazz`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    },
    answers: [{
      picture: `https://api.adorable.io/avatars/128/0`,
      artist: `John Snow`,
    }, {
      picture: `https://api.adorable.io/avatars/128/1`,
      artist: `Jack Daniels`,
    }, {
      picture: `https://api.adorable.io/avatars/128/2`,
      artist: `Jim Beam`,
    }],
  }
];

export const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export const SimpleMockComponent = () => <div />;
