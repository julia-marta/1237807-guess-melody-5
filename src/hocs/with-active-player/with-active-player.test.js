import React from 'react';
import renderer from 'react-test-renderer';
import withActivePlayer from "./with-active-player";
import {MockComponent} from "../../test-data";

const MockComponentWrapped = withActivePlayer(MockComponent);

it(`should withActivePlayer render correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
