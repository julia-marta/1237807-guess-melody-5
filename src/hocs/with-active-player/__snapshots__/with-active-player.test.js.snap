// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should withActivePlayer render correctly 1`] = `<div />`;
